sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumnsUtil.scala

```diff
@@ -284,16 +284,36 @@ object ResolveDefaultColumns extends QueryErrorsBase
         throw QueryCompilationErrors.defaultValuesUnresolvedExprError(
           statementType, colName, defaultSQL, ex)
     }
+    analyze(colName, dataType, parsed, defaultSQL, statementType)
+  }
+
+  def analyze(
+      colName: String,
+      dataType: DataType,
+      defaultValue: DefaultValue,
+      statementType: String): Expression = {
+    Option(defaultValue.getExpression)
+      .flatMap(ExpressionConverter.toV1)
+      .map(expr => analyze(colName, dataType, expr, expr.sql, statementType))
```

Comment:
nit: shall we use `Option(defaultValue.getSql()).getOrElse(expr.sql)` instead of `expr.sql`?